### ISSUE TITLE
Issue with XMLReporter - <system-out> field contains empty <![CDATA[]]>

### ISSUE THREAD

**Issue by misiekstewa** [testng-team/testng#2028]
### TestNG Version
6.14.3

### Expected behavior
Inside the TEST-mytestname.xml when it's running with JUnit 4

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuite errors="0" failures="0" hostname="mw" name="tests.external_frameworks.PetclinicTests" skipped="0" tests="3" time="0.155" timestamp="2019-03-08T08:48:00">
  <properties/>
  <testcase classname="tests.external_frameworks.PetclinicTests" name="validateSecurityLog" time="0.001" />
  <system-out><![CDATA[Starting process: /home/users/mmmm/builds/jdk-hs-8u171-linux-x64/bin/java -version
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b25)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b25, mixed mode)
Starting process: /home/users/mmmm/builds/jdk-hs-8u171-linux-x64/bin/java -version
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b25)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b25, mixed mode)
Starting process: /opt/wawawa/java/jdk-1.8.0/bin/java -version
openjdk version "1.8.0_71"
OpenJDK Runtime Environment (build 1.8.0_71-b15)
OpenJDK 64-Bit Server VM (build 25.71-b15, mixed mode)
Starting process: hostname
.
.
.
]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
```

### Actual behavior
Inside the TEST-mytestname.xml when it's running with TestNG

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<testsuite errors="0" failures="0" hostname="mw" name="tests.external_frameworks.PetclinicTests" skipped="0" tests="3" time="0.155" timestamp="2019-03-08T08:48:00">
  <properties/>
  <testcase classname="tests.external_frameworks.PetclinicTests" name="validateSecurityLog" time="0.001" />
  <system-out><![CDATA[]]></system-out>
  <system-err><![CDATA[]]></system-err>
</testsuite>
```

### Is the issue reproductible on runner?

- [ ] Shell
- [ ] Maven
- [x] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
I'm not sure if it's an issue or expected behaviour but when I'm running the same test with JUnit in the .xml report I'm able to see the system output, which is executed in static blocks before the test. 
So I mean everything which is appearing after:

```
> Task :Frameworks:test
Starting process 'Gradle Test Executor 1'. Working directory: /home/users/Projects/automation/Frameworks Command: /usr/lib/jvm/java-1.8.0-openjdk-1.8.0.71-1.b15.el6_7.x86_64/bin/java -Djava.security.manager=worker.org.gradle.process.internal.worker.child.BootstrapSecurityManager -Dorg.gradle.native=false -Dfile.encoding=UTF-8 -Duser.country=IE -Duser.language=en -Duser.variant -ea -cp /home/users/mmmm/.gradle/caches/4.8/workerMain/gradle-worker.jar worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Test Executor 1'
Successfully started process 'Gradle Test Executor 1'

Gradle Test Executor 1 STANDARD_OUT
```

but if I update annotations to TestNG annotations and update mu build.gradle file with

```
test {
    testLogging.showStandardStreams = true
    useTestNG() {
        useDefaultListeners = false
    }
}
```

then standard output is visible in the console but it's not saved in the .xml report file.

If that is a bug (not only issue with bad configuration) then I'll try to implement some small test which will be reproducing this.

**Comment by krmahadevan** [comment]
I don't think that the xml reports that TestNG generates ever captured the sys out and sys error in the reports. 

ping @cbeust @juherr WDYT ?

**Comment by juherr** [comment]
If possible, I think TestNG should have the same result than JUnit.
But I don't know what could be the technical impacts

**Comment by krmahadevan** [comment]
@juherr - I think it would be a challenge when trying to implement this for parallel execution, because the underlying PrintStream within System class dont seem to be thread safe.

**Comment by juherr** [comment]
Right, maybe the feature can be only activated when parallel is not activated.

Not a priority issue IMO.

**Comment by UrsLange** [comment]
I'd like to extend on this issue. I think the report, generated by `org.testng.reporters.JUnitReportReporter` should at least show the output, previously sent to `org.testng.Reporter`. This should be possible by writing the content of `Reporter.getOutput(currentTestResult)` to the `<system-out>` tag in the Report.
Compared to handling sysout and syserr, this sould be relatively easy to implement.
Thank you!

**Comment by UrsLange** [comment]
@krmahadevan May I ask you kindly, for feedback on my request? Implementing this would be a considerable help for my project. Thank you!

**Comment by krmahadevan** [comment]
@UrsLange - I don't want to club your ask in this issue, because this issue talks about capturing the `sys out` output, and you are asking for `Reporter.log()` contents. Please open up a new issue for your ask and I will try to get to it.

**Comment by UrsLange** [comment]
@krmahadevan Thank you for your response. I created #2124 for the matter.

**Comment by juherr** [comment]
AI triage verdict: **duplicate / superseded (external dependency)**.

The XML shown in the original reproducer is not produced by TestNG's `JUnitReportReporter`. The reproducer explicitly runs through Gradle with `useDefaultListeners = false`, and the shown report contains Gradle-style `<properties/>`, `<system-out>` and `<system-err>` elements. Current TestNG `JUnitReportReporter` instead writes its own reports under the `junitreports` directory, does not capture `System.out`/`System.err`, and only emits `<system-out>` from `org.testng.Reporter` output.

That distinction is also reflected in the later #2124: it separately requested `Reporter.log()` output in TestNG's own JUnit report, and that capability is present in the current reporter. It does not solve the original Gradle standard-stream capture scenario.

I found no `#2028`/`GITHUB-2028` code or `CHANGES.txt` reference and no open or merged TestNG PR implementing capture of Gradle's standard streams. The historical reproducer was **not executed** during this triage.

Because the failing `TEST-*.xml` in the reported configuration is generated by Gradle's test infrastructure, the actionable next step belongs in the Gradle/TestNG integration and Gradle XML reporting path rather than TestNG core.
